### PR TITLE
Add update api and function (including appending to list of comments)

### PR DESCRIPTION
**Pull request by jamiebrown201**
## Description

A function for updating any field in the database including appending a new comment onto the already existing list of comments. ive tried to make it as open as possible so when we add a field in a future debate we wont need to make code changes. Potential future work would be validation perhaps brining in a list of valid field through env variables.

## Implementation

Ive added an api route as well as the dynamodb method.

### CODE REVIEW

**Comment by jamiebrown201** [comment]
I also had to add prittier config because the linter wasn't liking a lot of the defaults.

File: models/dynamo_db.js [hunk 1]
```diff
@@ -189,6 +189,50 @@ async function getAllReports() {
...
+  try {
+    const params = {
+      Key: {
+        id: uuid,
+      },
+      ReturnValues: 'ALL_NEW',
+      ...updateExpressionConstruct(data),
+    };
+    const result = await query('update', params);
+    console.log(result);
```
Comment (Sqyphen) [inline models/dynamo_db.js]: Did you want to keep this console.log in? :)

File: models/dynamo_db.js [hunk 2]
```diff
@@ -189,6 +189,50 @@ async function getAllReports() {
...
+    return result.result;
+  } catch (err) {
+    console.error(err);
+    throw err;
```
Comment (Sqyphen) [inline models/dynamo_db.js]: How would we like to handle errors? Bubbling up a response to the function that called the dynamoDB func? or have the function throw the err and stop?

File: models/dynamo_db.js [hunk 3]
```diff
@@ -189,6 +189,50 @@ async function getAllReports() {
...
+  let expressionAttributeValues = {};
+  let updateExpression = 'SET';
+  const fields = Object.keys(data);
+  fields.forEach((key, index) => {
+    expressionAttributeValues = {
+      ...expressionAttributeValues,
+      [`:${key}`]: data[key],
+    };
+    if (LIST_TYPES.includes(key)) {
+      updateExpression =
```
Comment (Sqyphen) [inline models/dynamo_db.js]: Could you use `updatedExpression +=` here?

File: models/dynamo_db.js [hunk 4]
```diff
@@ -189,6 +189,50 @@ async function getAllReports() {
...
+    const result = await query('update', params);
+    console.log(result);
+    return result.result;
+  } catch (err) {
+    console.error(err);
+    throw err;
+  }
+}
+
+function updateExpressionConstruct(data) {
```
Comment (Sqyphen) [inline models/dynamo_db.js]: nice function

File: models/dynamo_db.js (hunk 1 above)
Comment (jamiebrown201) [reply models/dynamo_db.js]: Boo no :(

File: models/dynamo_db.js (hunk 2 above)
Comment (jamiebrown201) [reply models/dynamo_db.js]: Im really not sure what the best way to proceed with errors for now. What do you think?

Comment (jamiebrown201) [reply models/dynamo_db.js]: Of course we will come up with a sustainable solution at some point but you are right we should probably standardise what we do now.

File: models/dynamo_db.js (hunk 3 above)
Comment (jamiebrown201) [reply models/dynamo_db.js]: I thought this was just a ruby thing! 😱

File: models/dynamo_db.js (hunk 4 above)
Comment (jamiebrown201) [reply models/dynamo_db.js]: 😄

File: models/dynamo_db.js (hunk 3 above)
Comment (Sqyphen) [reply models/dynamo_db.js]: Nope! :) But I know some people prefer not to use it

**Pull request merged**